File: lib/swig/cs/NatureListComboBox.cs
/*
 * Copyright (c) 2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

using System.Windows.Forms;

namespace PKMN
{
    namespace Forms
    {
        public class NatureListComboBox: ComboBox
        {
            public NatureListComboBox(): base()
            {
                StringVector nature_vec = new StringVector();
                Lists.getNatureList(nature_vec);
                for(int i = 0; i < nature_vec.Count; i++)
                {
                    Items.Add(nature_vec[i]);
                }
            }
        }
    }
}

File: lib/swig/java/NatureListComboBox.java
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package org.nc.LibPKMN;

import javax.swing.*;

public class NatureListComboBox extends JComboBox<String>
{
    public NatureListComboBox(int gen)
    {
        super();

        StringVector natureVec = new StringVector();
        LibPKMN.getNatureList(natureVec);
        for(int i = 0; i < natureVec.size(); i++)
        {
            addItem(natureVec.get(i));
        }

        setEditable(false);
    }
}
